### DISCUSSION TITLE
Stamp tool

### DISCUSSION THREAD

**Discussion by elvisv8** [stephanrauh/ngx-extended-pdf-viewer#1824] [Q&A]
Thanks for adding the stamp tool, it's a very useful feature. But I noticed that there is no way to resize the uploaded picture (only the width). In pdf.js resizers elements are used for this purpose:

```html
<div class="resizers">
  <div class="resizer topLeft"></div>
  <div class="resizer topRight"></div>
  <div class="resizer bottomRight"></div>
  <div class="resizer bottomLeft"></div>
  <div class="resizer topMiddle"></div>
  <div class="resizer middleRight"></div>
  <div class="resizer bottomMiddle"></div>
  <div class="resizer middleLeft"></div>
</div>
```

And it looks like this:
![image](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/74044604/e183e73b-d271-47fd-8030-b2b74f6f0992)

A quick review showed that there are no styles for them. I would be glad to solve this problem by common forces.

**Comment by stephanrauh** [comment]
Please do! I'm looking forward to your pull request!

You find the CSS rules in the folder `/projects/ngx-extended-pdf-viewer/src/lib/theme/common`. There you find a couple of *.scss files, which are derived from the corresponding CSS files of pdf.js. I try to make my files resemble the original files as much as I can, so I can simply compare my SCSS files to the original CSS files. Sounds complicated, but I guess you get the idea quickly if you compare the two sets of files in a diff tool.

The missing CSS rules are probably here: https://github.com/mozilla/pdf.js/blob/master/web/annotation_editor_layer_builder.css

To build the library locally, you can use the script `npm run showcase` after modifying the `package.json`. The `showcase` script compiles the library and copies it to a my showcase project. Modify the `cp` statement, so it copies the library to the node_modules folder of your own project:
![grafik](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/3045767/098e48e5-2fcd-4cc3-8ca5-8dde0d676e70)

If you're using Windows, I hope Git bash helps you. There's also the `win-package` script that's the Windows counterpart of `unix-package`, but I haven't used it for more than a year, so I'm afraid it's broken by now.

Did I forget anything? Oh, yes, one more thing: Happy coding!

Stephan

**Reply by stephanrauh** [reply]
I knew I'd forgotten something. If you've using Angular 16, you must delete the `.angular` folder after running `npm run showcase`. Angular caches aggressivly, so it misses the modified node_modules folder.

**Reply by stephanrauh** [reply]
Version 18.0.0-beta.5 is going to solve the problem. As usual, no promises, but I suppose I'm going to publish the new version this week-end.
